Server/src/models/GraphsModel.ts

```diff
@@ -152,38 +185,53 @@ export class GraphsModel {
-        let axisLog: boolean[] = []
-        let axisZoomStart: number[] = []
-        let axisZoomEnd: number[] = []
-        let axisUnits: string[] = []
-        for (let j = 0; j < graph.datasets.length; j++) {
-            datasetIds.push(graph.datasets[j].id)
-            datasetColors.push(graph.datasets[j].color)
-            datasetShapes.push(graph.datasets[j].shape)
-            datasetHiddenStatus.push(graph.datasets[j].isHidden)
+    private updateGraphQuery = (graphId: number, graphName: string, datasetIds: number[], datasetColors: string[], datasetShapes: string[],
+        datasetHiddenStatus: boolean[], axisVariable: string[], axisLog: boolean[], axisZoomStart: number[], axisZoomEnd: number[], axisUnits: string[]) =>
+        this.connection.createQueryBuilder()
+            .update(Graphstate)
+            .set({
+                name: graphName, datasetIds: datasetIds,
+                datasetColors: datasetColors, datasetShapes: datasetShapes,
+                datasetHiddenStatus: datasetHiddenStatus,
+                axisVariable: axisVariable, axisLog: axisLog,
+                axisZoomStart: axisZoomStart, axisZoomEnd: axisZoomEnd,
+                axisUnits: axisUnits
+            })
+            .where("id = :id", { id: graphId })
+            .execute();
+
+    async updateGraph(graph: IGraphStateModel, userId: number): Promise<any[]> {
+        let statusCode = 500
+        let statusMessage = "Something went wrong fetching from DB. Maybe its down"
+        let graphOwner = await selectGraphStateAccountQuery(this.connection, Number(graph.id))
+        if (graphOwner == undefined) {
+            statusCode = 400
+            statusMessage = "Graph does not exist"
         }
-        for (let k = 0; k < graph.axes.length; k++) {
-            axisVariable.push(graph.axes[k].variableName)
-            axisLog.push(graph.axes[k].logarithmic)
-            axisZoomStart.push(graph.axes[k].zoomStartIndex)
-            axisZoomEnd.push(graph.axes[k].zoomEndIndex)
-            axisUnits.push(graph.axes[k].units)
+        else if (graphOwner.accountId !== userId) {
+            statusCode = 400
+            statusMessage = "This is not your graph!"
         }
-        let statusMessage = await this.sendSavedGraphToDatabase(userId, graph.name, datasetIds, datasetColors, datasetShapes, datasetHiddenStatus, axisVariable, axisLog, axisZoomStart, axisZoomEnd, axisUnits)
-        return statusMessage
-
+        else {
+            let processGraphInput = await this.processGraphInput(graph)
+            await this.updateGraphQuery(processGraphInput[0], processGraphInput[1], processGraphInput[2], processGraphInput[3], processGraphInput[4],
+                processGraphInput[5], processGraphInput[6], processGraphInput[7], processGraphInput[8], processGraphInput[9], processGraphInput[10])
+            statusCode = 200
+            statusMessage = "Graph successfully updated"
+        }
+        return [statusCode, statusMessage]

```

Comment:
We can keep the return value to be 1 attribute by moving some of this logic outside

Reply:
See above